src/Core/Content/Flow/Dispatching/BufferedFlowExecutor.php

```diff
@@ -0,0 +1,132 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Shopware\Core\Content\Flow\Dispatching;
+
+use Doctrine\DBAL\Connection;
+use Psr\Container\ContainerInterface;
+use Shopware\Core\Content\Flow\Exception\ExecuteSequenceException;
+use Shopware\Core\Framework\Event\FlowEventAware;
+use Shopware\Core\Framework\Feature;
+use Shopware\Core\Framework\Log\Package;
+use Symfony\Component\EventDispatcher\EventSubscriberInterface;
+use Symfony\Component\HttpKernel\KernelEvents;
+use Symfony\Contracts\Service\ServiceSubscriberInterface;
+
+/**
+ * @internal not intended for decoration or replacement
+ */
+#[Package('services-settings')]
+class BufferedFlowExecutor implements EventSubscriberInterface, ServiceSubscriberInterface
+{
+    /**
+     * @var array<FlowEventAware>
+     */
+    private array $bufferedEvents = [];
+
+    public function __construct(
+        private readonly ContainerInterface $container,
+    ) {
+    }
+
+    public static function getSubscribedEvents(): array
+    {
+        if (!Feature::isActive('v6.7.0.0')) {
+            return [];
+        }
+
+        return [
+            BufferFlowExecutionEvent::class => 'handleBufferFlowExecutionEvent',
+            KernelEvents::TERMINATE => 'executeBufferedEvents',
+        ];
+    }
+
+    public function handleBufferFlowExecutionEvent(BufferFlowExecutionEvent $bufferFlowExecutionEvent): void
+    {
+        $this->bufferedEvents[] = $bufferFlowExecutionEvent->getEvent();
+    }
+
+    public function executeBufferedEvents(): void
+    {
+        do {
+            $events = $this->bufferedEvents;
+            $this->bufferedEvents = [];
+            $flowLoader = $this->container->get(FlowLoader::class);
+            $flows = $flowLoader->load();
+
+            foreach ($events as $event) {
+                $storableFlow = $this->container->get(FlowFactory::class)->create($event);
+                $this->callFlowExecutor($storableFlow, $flows);
+            }
+        } while (!empty($this->bufferedEvents));
```

Comment:
What does the outer loop do? Is it for executing events that are triggered during the first chunk of events (like flow triggers another flow?). In that case a comment would be great. We might also want to prevent infinite loops here and add an upper bound before reporting an error 🤔 

 Also you could store the container lookups in variables, my suggestion would be something like this:
```suggestion
        $flowLoader = $this->container->get(FlowLoader::class);
        $flowFactory = $this->container->get(FlowFactory::class);
        $batchCounter = 0;
        
        do {
            $events = $this->bufferedEvents;
            $this->bufferedEvents = [];
            $flows = $flowLoader->load();

            foreach ($events as $event) {
                $storableFlow = $flowFactory->create($event);
                $this->callFlowExecutor($storableFlow, $flows);
            }
        } while (!empty($this->bufferedEvents) && $batchCounter < 10);
        
        if ($batchCounter >= 10) {
            // report recursion limit exceeded error
        }
```

Reply:
That's correct, good point about the 'recursion' limit. That would actually be an improvement over the old system, where as far as i can tell such an infinite loop would have crashed the application. I will add a comment as well.